src/components/elements/wrapper/conditional-renderer.tsx

```diff
@@ -80,8 +85,16 @@ export const ConditionalRenderer = ({ id, renderRules, children, schema }: IProp
 	/**
 	 * Return all object keys by recursively looping through the object children
 	 */
-	const listAllChildIds = (children: unknown) => {
+	const listAllChildIds = (schema: IProps["schema"]) => {
+		const children = schema["children"];
 		const childIdList: string[] = [];
+
+		// Handle special fields that render additional fields
+		if (schema.uiType === "chips") {
+			childIdList.push(id + "-textarea");
+		}
```

Comment:
why add chips textarea if the actual unregister is in chips component itself?

Reply:
`Chips` cannot handle the case where it is nested and hidden, because it doesn't get mounted. So `ConditionalRenderer` has to handle this since it is aware of nested fields.